### PR TITLE
Improve display of builtin types by the toplevel

### PR DESCRIPTION
**Pull request by samsa1**
Currently the toplevel uses type information to know how to display types such as lists.
However this works only if the type is exactly the builtin types and not an alias.

Thus the some examples are not displayed as nicely as they could :
```ocaml
# let f (l : 'a list) : 'a List.t = l
  ;;
val f : 'a list -> 'a List.t = <fun>
# f [3; 2];;
- : int List.t = List.(::) (3, [2])
```

The change proposed by this PR is to check if a type correspond to a builtin type through aliases to detect more occurrences of builtin types.

### CODE REVIEW

**Comment by samsa1** [comment]
Sorry I had a problem of mixed commits with another PR that is already merged. I'll try to clean up the diffs

File: toplevel/genprintval.ml [hunk 1]
```diff
@@ -264,14 +264,14 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
         try
           find_printer depth env ty obj
         with Not_found ->
-          match get_desc ty with
-          | Tvar _ | Tunivar _ ->
+          match get_desc ty, get_desc (Ctype.expand_head_opt env ty) with
```
Comment (Octachron) [inline toplevel/genprintval.ml]: That's looks like a good idea, but I would propose to move this expansion inside the `Tconstr` case. Also, it should be better to use the public version `expand_head` rather the optimisation-oriented `expand_head_opt`.

**Comment by samsa1** [comment]
Ok, I managed to clean up the code, this should help reviewing

File: toplevel/genprintval.ml (hunk 1 above)
Comment (samsa1) [reply toplevel/genprintval.ml]: Done

**Review by Octachron — APPROVED** [review]
This is a clear improvement with a minimal code complexity cost. Even if not all possible cases are not handled, the more confusing one should be covered. In particular, using
```ocaml
open List
```
doesn't break the toplevel list printer anymore.

Thus I think it is better to merge this PR. Thanks!

**Pull request merged**